addons/mail/static/tests/chat_window/chat_window.test.js

```diff
@@ -1187,3 +1060,39 @@ test("hiding/swapping hidden chat windows does not update server state", async (
     await contains(".o-mail-ChatWindow", { text: "Sales" });
     await assertSteps([]);
 });
+
+test("mark as read when opening chat window", async () => {
```

Comment:
Name of test is too broad for what it tests.

I don't mind just naming of test if it was grouping all assertions related to this feature. But that's not how we make our tests and instead we have many small tests. Thus each test are specific and the name should be as specific as necessary for the good understanding of its intended scope.

Here I suppose _"close chat window should mark thread as read"_ is the intended test coverage?

Reply:
The issue is always the same with our tests: we need some positive assertions. Here, we need to assert that the counter is shown. Then we can finally test what the test was created for: the fact that this counter  disappears when opening the chat window. The first part is just setup: we open the chat window, we receive a message, the counter appears. The second part is what matters (hence the name): we open the same chat window which is marked as read thus the counter disappears.